controllers/goroutines/cleanup_deprecated_services.go

```diff
@@ -0,0 +1,225 @@
+		},
+		{
+			name:  "monitoring-exporters-operator-request",
+			group: "operator.ibm.com",
+			kind:  "OperandRequest",
+		},
+	},
+	"ibm-monitoring-prometheusext-operator": {
+		{
+			name:  "ibm-monitoring",
+			group: "monitoring.operator.ibm.com",
+			kind:  "PrometheusExt",
+		},
+		{
+			name:  "monitoring-prometheus-ext-operator-request",
+			group: "operator.ibm.com",
+			kind:  "OperandRequest",
+		},
+	},
+	"ibm-metering-operator": {
+		{
+			name:  "metering",
+			group: "operator.ibm.com",
+			kind:  "Metering",
+		},
+		{
+			name:  "meteringui",
+			group: "operator.ibm.com",
+			kind:  "MeteringUI",
+		},
+		{
+			name:  "meteringreportserver",
+			group: "operator.ibm.com",
+			kind:  "MeteringReportServer",
+		},
+		{
+			name:  "ibm-metering-bindinfo",
+			group: "operator.ibm.com",
+			kind:  "OperandBindInfo",
+		},
+		{
+			name:  "ibm-metering-request",
+			group: "operator.ibm.com",
+			kind:  "OperandRequest",
+		},
+	},
+	"ibm-elastic-stack-operator": {
+		{
+			name:  "logging",
+			group: "elasticstack.ibm.com",
+			kind:  "ElasticStack",
+		},
+		{
+			name:  "ibm-elastic-stack-bindinfo",
+			group: "operator.ibm.com",
+			kind:  "OperandBindInfo",
+		},
+		{
+			name:  "ibm-elastic-stack-request",
+			group: "operator.ibm.com",
+			kind:  "OperandRequest",
+		},
+	},
+	"ibm-catalog-ui-operator": {
+		{
+			name:  "catalog-ui",
+			group: "operator.ibm.com",
+			kind:  "CatalogUI",
+		},
+		{
+			name:  "catalog-ui-request",
+			group: "operator.ibm.com",
+			kind:  "OperandRequest",
+		},
+	},
+	"ibm-helm-api-operator": {
+		{
+			name:  "helm-api",
+			group: "operator.ibm.com",
+			kind:  "HelmAPI",
+		},
+		{
+			name:  "helm-api-request",
+			group: "operator.ibm.com",
+			kind:  "OperandRequest",
+		},
+	},
+	"ibm-helm-repo-operator": {
+		{
+			name:  "helm-repo",
+			group: "operator.ibm.com",
+			kind:  "HelmRepo",
+		},
+		{
+			name:  "helm-repo-request",
+			group: "operator.ibm.com",
+			kind:  "OperandRequest",
+		},
+	},
+}
+
+// CleanUpDeprecatedServices will clean up deprecated services' CRD, operandBindInfo, operandRequest, subscription, CSV
+func CleanUpDeprecatedServices(bs *bootstrap.Bootstrap) {
+	for {
+		for service, resourcesList := range deprecatedServicesMap {
+			for _, resource := range resourcesList {
+				operatorNs, err := util.GetOperatorNamespace()
+				if err != nil {
+					continue
+				}
+
+				if err := cleanup(bs, resource.name, operatorNs, resource.group, resource.kind); err != nil {
+					continue
+				}
+			}
+
+			// delete sub & csv
+			if err := deleteSubscription(bs, service, MasterNamespace); err != nil {
+				klog.Errorf("Delete subscription failed: %v", err)
+				continue
+			}
+		}
+
+		time.Sleep(2 * time.Minute)
+	}
+}
+
+func cleanup(bs *bootstrap.Bootstrap, name, operatorNs string, group, kind string) error {
+	resource := &unstructured.Unstructured{}
+	resource.SetGroupVersionKind(schema.GroupVersionKind{Group: group, Version: "v1alpha1", Kind: kind})
```

Comment:
@ZhuoxiLi Should we add the version into the resource struct as well?

Reply:
sure, I will add the version attribute.